### PR TITLE
[IRRC-89] Create page to show Teacher profile

### PR DESCRIPTION
**Pull request by lauratsui**
Created new TeacherProfile.vue. WIP

### CODE REVIEW

**Review by aelbore — APPROVED** [review]
Looks good

File: frontend/src/pages/StudentProfile.vue [hunk 1]
```diff
@@ -77,23 +92,24 @@ export default {
...
+      nativeLanguage: data.nativeLanguage,
+      status: data.status,
+      Notes: data.Notes,
+      dateJoined: new Date(data.created_at).toDateString(),
+      EnglishProficiency: data.EnglishProficiency,
+    };
+
 
     const studentHistory = data.statusUpdates.map(update => {
       if (update.reason == null) {
```
Comment (therealyingtong) [inline frontend/src/pages/StudentProfile.vue:114]: Is the Student Profile (e.g. `localhost:3000/students/1` working for you? I had to make the below change to get it working:

```suggestion
      update.reason = Object.assign([], update.reason)

      if (update.reason.Reason === undefined) {
```

**Review by therealyingtong — CHANGES_REQUESTED** [review]

**Review by therealyingtong — APPROVED** [review]

**Review by therealyingtong — APPROVED** [review]

File: frontend/src/components/StudentProfileCard.vue [hunk 2]
```diff
@@ -3,7 +3,8 @@
     <div class="edit-container">Edit</div>
     <div class="student-grid">
       <img src="../assets/images/student.png" />
```
Comment (dzldebby) [inline frontend/src/components/StudentProfileCard.vue]: Image displayed should switch on `isTeacher` boolean to display `teacher.png`

File: frontend/src/pages/TeacherProfile.vue [hunk 3]
```diff
@@ -0,0 +1,164 @@
+<template>
+  <Page blueBg>
+    <div class="teacher-profile-container">
+      <div class="teacher-profile-left">
+        <StudentProfileCard
```
Comment (dzldebby) [inline frontend/src/pages/TeacherProfile.vue:5]: Let's rename reused components to be more generic. See #87

File: frontend/src/pages/TeacherProfile.vue [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        <section class="student-status-section">
+          <StatusCardScreening
+            :studentID="teacherID"
+            :englishProficiency="englishProficiency"
+            v-if="teacherData.status.Description.toUpperCase() === 'SCREENING'"
+          />
+          <StatusCardUnmatched
+            :studentID="teacherID"
+            v-if="teacherData.status.Description.toUpperCase() === 'UNMATCHED'"
+          />
```
Comment (therealyingtong) [inline frontend/src/pages/TeacherProfile.vue]: In `StatusCardUnmatched.vue`, `studentID` is passed to `ModalDroppedOut.vue`, which calls `updateStudentStatus(studentID)`. If we pass `teacherID`, this attempts to call `updateStudentStatus(teacherID)`, which attempts to update the student status using the teacher ID. This causes an update conflict.

To test this, observe that clicking "Drop Out" on the Teacher Profile does not change the teacher status.

**Review by therealyingtong — CHANGES_REQUESTED** [review]

File: frontend/src/pages/TeacherProfile.vue (hunk 4 above)
Comment (therealyingtong) [reply frontend/src/pages/TeacherProfile.vue]: Instead, we need `StatusCardUnmatched` and `ModalDroppedOut` to switch on `isTeacher` as well.

Comment (dzldebby) [inline frontend/src/pages/TeacherProfile.vue]: Unlike in the `StudentProfile.vue`, in `TeacherProfile.vue` we should only be able to select "Drop out" for an unmatched Teacher. We shouldn't be able to select "Select Match". See screenshots below from Figma. So we'll need to switch on `isTeacher` in the `StatusCardUnmatched` component. 

image

image

**Review by dzldebby — CHANGES_REQUESTED** [review]
We would suggest to keep this PR front-end related :) And the issues with the switching logic can be addressed in another PR. See also #87

File: frontend/src/pages/TeacherProfile.vue (hunk 4 above)
Comment (lauratsui) [reply frontend/src/pages/TeacherProfile.vue]: Thanks! let me make those changes

Comment (lauratsui) [reply frontend/src/pages/TeacherProfile.vue]: Thanks! We've made the changes to ModalDroppedOut.vue, as well as the status cards for Dropped Out, Matched and Unmatched.

File: frontend/src/pages/TeacherProfile.vue (hunk 3 above)
Comment (lauratsui) [reply frontend/src/pages/TeacherProfile.vue:5]: Noted, we can do the renaming to another PR

File: frontend/src/components/StudentProfileCard.vue (hunk 2 above)
Comment (lauratsui) [reply frontend/src/components/StudentProfileCard.vue]: Good catch, made the changes thanks!

File: frontend/src/pages/TeacherProfile.vue (hunk 4 above)
Comment (lauratsui) [reply frontend/src/pages/TeacherProfile.vue]: Thanks, we made the changes to the cards!

**Review by dzldebby — APPROVED** [review]
LGTM

**Pull request merged**